Flapjack/CoreData/CoreDataSource.swift

```diff
@@ -168,6 +179,8 @@ public class CoreDataSource<T: NSManagedObject & DataObject>: NSObject, NSFetche
     public func endListening() {
         guard hasExecuted else { return }
         controller.delegate = nil
+        NotificationCenter.default.removeObserver(self, name: CoreDataAccess.didCreateNewMainContextNotification, object: nil)
+        NotificationCenter.default.removeObserver(self, name: CoreDataAccess.willDestroyMainContextNotification, object: nil)
```

Comment:
Would be interested to know if any of our client code actually calls this.

Reply:
Calls `endListening`, you mean?

Reply:
Yes.

Reply:
In a couple places yeah — notably around user history (which I believe you wrote, so thank you for being a good citizen).

Reply:
I think that was the first time I used the new core data sources from flapjack.  We should make sure that all other uses of them in the app also call `endListening`.